tests/MongoDB.Driver.Core.Tests/Core/Operations/OperationTestBase.cs

```diff
@@ -398,61 +400,111 @@ protected void Update(string filter, string update)
             Func<WritableServerBinding, CancellationToken, Task<TResult>> executeAsync,
             Func<WritableServerBinding, CancellationToken, TResult> execute,
+            Action<EventCapturer, ICoreSessionHandle, Exception> assertResults,
             string commandName,
-            bool async)
+            bool async,
+            bool useExplicitSessionIfSupported = true)
         {
             var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName == commandName);
             using (var cluster = CoreTestConfiguration.CreateCluster(b => b.Subscribe(eventCapturer)))
             {
-                using (var session = CoreTestConfiguration.StartSession(cluster))
+                using (var session = CreateSession(useExplicitSessionIfSupported, cluster))
                 using (var binding = new WritableServerBinding(cluster, session.Fork()))
                 {
                     var cancellationToken = new CancellationTokenSource().Token;
+                    Exception exception;
                     if (async)
                     {
-                        executeAsync(binding, cancellationToken).GetAwaiter().GetResult();
+                        exception = Record.Exception(() => executeAsync(binding, cancellationToken).GetAwaiter().GetResult());
                     }
                     else
                     {
-                        execute(binding, cancellationToken);
+                        exception = Record.Exception(() => execute(binding, cancellationToken));
                     }
 
-                    var commandStartedEvent = (CommandStartedEvent)eventCapturer.Next();
-                    var command = commandStartedEvent.Command;
-                    if (session.Id == null)
-                    {
-                        command.Contains("lsid").Should().BeFalse();
-                    }
-                    else
-                    {
-                        command["lsid"].Should().Be(session.Id);
-                    }
-                    session.ReferenceCount().Should().Be(2);
+                    assertResults(eventCapturer, session, exception);
                 }
             }
         }
 
+        // private methods
+        private void AssertSessionIdIsSentIfPresented(EventCapturer eventCapturer, ICoreSessionHandle session, Exception exception)
+        {
+            exception.Should().BeNull();
+            var commandStartedEvent = (CommandStartedEvent)eventCapturer.Next();
+            var command = commandStartedEvent.Command;
+            if (session.Id == null)
+            {
+                command.Contains("lsid").Should().BeFalse();
+            }
+            else
+            {
+                command["lsid"].Should().Be(session.Id);
+            }
+
+            session.ReferenceCount().Should().Be(2);
+        }
+
+        private void AssertSessionIdWithUnacknowledgedWrite(EventCapturer eventCapturer, ICoreSessionHandle session, Exception ex)
```

Comment:
Rename to `AssertSessionIdWasNotSentIfUnacknowledgedWrite` to match the method this was called from.